drivers/media/i2c/mira016.inl

```diff
@@ -0,0 +1,6501 @@
+ * To change the TBD I2C address, set bit 6&7 to 0b10,
+ * then the reg_val will become TBD I2C address.
+ * The TBD I2C address is stored in mira016->tbd_client_i2c_addr.
+ */
+#define AMS_CAMERA_CID_MIRA016_REG_FLAG_I2C_SEL 0b01100000
+#define AMS_CAMERA_CID_MIRA016_REG_FLAG_I2C_MIRA 0b00000000
+#define AMS_CAMERA_CID_MIRA016_REG_FLAG_I2C_TBD 0b00100000
+#define AMS_CAMERA_CID_MIRA016_REG_FLAG_I2C_SET_TBD 0b01000000
+
+/* Pre-allocated i2c_client */
+#define MIRA016PMIC_I2C_ADDR 0x2D
+#define MIRA016UC_I2C_ADDR 0x0A
+#define MIRA016LED_I2C_ADDR 0x53
+
+#define MIRA016_NATIVE_WIDTH 400U
+#define MIRA016_NATIVE_HEIGHT 400U
+
+#define MIRA016_PIXEL_ARRAY_LEFT 0U
+#define MIRA016_PIXEL_ARRAY_TOP 0U
+#define MIRA016_PIXEL_ARRAY_WIDTH 400U
+#define MIRA016_PIXEL_ARRAY_HEIGHT 400U
+
+/* Set analog gain min and max to 0 to avoid user changing it. */
+#define MIRA016_ANALOG_GAIN_MAX 1
+#define MIRA016_ANALOG_GAIN_MIN 0
+#define MIRA016_ANALOG_GAIN_STEP 1
+#define MIRA016_ANALOG_GAIN_DEFAULT MIRA016_ANALOG_GAIN_MIN
+
+#define MIRA016_BANK_SEL_REG 0xE000
+#define MIRA016_RW_CONTEXT_REG 0xE004
+#define MIRA016_CMD_REQ_1_REG 0x000A
+#define MIRA016_CMD_HALT_BLOCK_REG 0x000C
+
+// Exposure time is indicated in us
+#define MIRA016_EXP_TIME_L_REG 0x000E
+#define MIRA016_EXP_TIME_S_REG 0x0012
+
+// Target frame time is indicated in us
+#define MIRA016_TARGET_FRAME_TIME_REG 0x0008
+
+#define MIRA016_SUPPORTED_XCLK_FREQ 24000000
+
+// Some timings
+#define MIRA016_DATA_RATE 1500 // Mbit/s
+#define MIRA016_SEQ_TIME_BASE 8 / MIRA016_DATA_RATE
+#define MIRA016_LPS_CYCLE_TIME 1145
+#define MIRA016_GLOB_TIME 68
+#define MIRA016_ROW_LENGTH 1504 // 12b
+#define MIRA016_LPS_DISABLED 0
+#define MIRA016_TROW_US MIRA016_ROW_LENGTH * 8 / MIRA016_DATA_RATE
+
+#define MIRA016_READOUT_TIME MIRA016_TROW_US * (11 + MIRA016_PIXEL_ARRAY_HEIGHT)
+
+// Default exposure is adjusted to 1 ms
+#define MIRA016_LUT_DEL_008 0
+#define MIRA016_GRAN_TG 1500 * 50 / MIRA016_DATA_RATE
+#define MIRA016_MIN_ROW_LENGTH MIRA016_ROW_LENGTH // 1042 for 8 bit
+#define MIRA016_MIN_ROW_LENGTH_US (MIRA016_MIN_ROW_LENGTH * 8 / MIRA016_DATA_RATE)
+#define MIRA016_EXPOSURE_MIN_US (int)(1 + (151 + MIRA016_LUT_DEL_008) * MIRA016_GRAN_TG * 8 / MIRA016_DATA_RATE)
+#define MIRA016_EXPOSURE_MAX_US (1000000)
+#define MIRA016_DEFAULT_EXPOSURE_US 1000
+// Default exposure for V4L2 is in row time
+
+// #define MIRA016_MIN_VBLANK 11 // for 10b or 8b, 360fps
+#define MIRA016_MIN_VBLANK_200 4610 // 200 fps
+#define MIRA016_MIN_VBLANK_360 2400  // 200 fps
+#define MIRA016_MAX_VBLANK	   1000000
+
+#define MIRA016_DEFAULT_VBLANK_60 16000  // 200 fps
+
+// Power on function timing
+#define MIRA016_XCLR_MIN_DELAY_US 150000
+#define MIRA016_XCLR_DELAY_RANGE_US 3000
+
+// pixel_rate = link_freq * 2 * nr_of_lanes / bits_per_sample
+// 0.9Gb/s * 2 * 1 / 12 = 157286400
+// 1.5 Gbit/s * 2 * 1 / 12 = 250 000 000
+#define MIRA016_PIXEL_RATE (400000000)
+/* Should match device tree link freq */
```

Comment:
It should also be correct for the sensor. Below you reference `MIRA016_DATA_RATE 1500 //Mbit/s`, which would imply the link frequency is 750MHz 

NB technically this exceeds the spec for the Pi0-4 CSI2 receiver, however you may get away with it.